api/admin/controller.py

```diff
@@ -419,55 +320,178 @@ def resolve_complaints(self, data_source, identifier):
             if c.subject.genre
         ]
+        
+        # Update audience
+        new_audience = flask.request.form.get("audience")
+        if new_audience != work.audience:
+            # Delete all previous staff audience classifications
+            for c in old_classifications:
+                if c.subject.type == Subject.FREEFORM_AUDIENCE:
+                    self._db.delete(c)
+
+            # Create a new classification with a high weight
+            primary_identifier.classify(
+                data_source=staff_data_source,
+                subject_type=Subject.FREEFORM_AUDIENCE,
+                subject_identifier=new_audience,
+                weight=WorkController.STAFF_WEIGHT,
+            )
+
+        # Update target age if present
+        new_target_age_min = flask.request.form.get("target_age_min")
+        new_target_age_max = flask.request.form.get("target_age_max")
+        if new_target_age_max < new_target_age_min:
+            return INVALID_EDIT.detailed("Minimum target age must be less than maximum target age.")
+
+        if work.target_age:
+            old_target_age_min = work.target_age.lower
+            old_target_age_max = work.target_age.upper
+        else:
+            old_target_age_min = None
+            old_target_age_max = None
+        if new_target_age_min != old_target_age_min or new_target_age_max != old_target_age_max:
+            # Delete all previous staff target age classifications
+            for c in old_classifications:
+                if c.subject.type == Subject.AGE_RANGE:
+                    self._db.delete(c)
+
+            # Create a new classification with a high weight - higher than audience
+            if new_target_age_min and new_target_age_max:
+                age_range_identifier = "%s-%s" % (new_target_age_min, new_target_age_max)
+                primary_identifier.classify(
+                    data_source=staff_data_source,
+                    subject_type=Subject.AGE_RANGE,
+                    subject_identifier=age_range_identifier,
+                    weight=WorkController.STAFF_WEIGHT * 100,
+                )
+
+        # Update fiction status
+        new_fiction = True if flask.request.form.get("fiction") == "fiction" else False
+        if new_fiction != work.fiction:
+            # Delete previous staff fiction classifications
+            for c in old_classifications:
+                if c.subject.type == Subject.SIMPLIFIED_FICTION_STATUS:
+                    self._db.delete(c)
+
+            # Create a new classification with a high weight (higher than genre)
+            fiction_term = "Fiction" if new_fiction else "Nonfiction"
+            classification = primary_identifier.classify(
+                data_source=staff_data_source,
+                subject_type=Subject.SIMPLIFIED_FICTION_STATUS,
+                subject_identifier=fiction_term,
+                weight=WorkController.STAFF_WEIGHT * 100,
```

Comment:
now that staff can't create incompatible fiction/genre combinations, fiction classifications shouldn't need the extra high weight.